tools/automator-main.sh

```diff
@@ -281,7 +290,14 @@ commit() {
 merge() {
   local src_branch="${AUTOMATOR_SRC_BRANCH:-none}"
   fork_name="$src_branch-$branch-$modifier-$(hash "$title")"
-  git remote add -f -t "$merge_branch" upstream "$merge_repository"
+
+  if $fetch_tags; then
+    merge_branch="$(git describe --tags --abbrev=0 "$merge_branch")"
```

Comment:
this `if` statement could end here.

That's all you need. The `git remote add` command from above goes below.

In code:
```sh
if $fetch_tags; then
  merge_branch="$(git describe --tags --abbrev=0 "$merge_branch")"
fi
git remote add -f -t "$merge_branch" upstream "$merge_repository"
```

```

Reply:
Done. But it is failing on my machine (maybe because of my forks):

```
+ true
+ '[' -n maistra-2.3 ']'
++ git describe --tags --abbrev=0 maistra-2.3
+ merge_branch=maistra-2.3.1
+ git remote add -f -t maistra-2.3.1 upstream https://github.com/leoieggli/api
Updating upstream
fatal: couldn't find remote ref refs/heads/maistra-2.3.1
error: Could not fetch upstream
```

Reply:
If it's failing the it's not correct. Please schedule a call so we can test this together.

Reply:
Fixed. The script worked - the git remote add is a little bit different  when using tags. So, the code will be like this:

`  if $fetch_tags; then
    if [ "${merge_strategy}" == "merge" ]; then
      merge_branch="$(git describe --tags --abbrev=0 "$merge_branch")"
      git remote add upstream $merge_repository
      git fetch upstream tag $merge_branch
      echo "Using tag $merge_branch"
    else
      print_error_and_exit "merge_strategy value must be 'merge', and is required when using '--fetch-tags' option"
    fi
  else
    git remote add -f -t "$merge_branch" upstream "$merge_repository"
    echo "Using branch $merge_branch"
  fi`